ToDo.Core/ViewModels/ToDoViewModel.cs

```diff
@@ -160,7 +285,32 @@ internal async Task SearchedTextChanged(string query)
                 return;
             }
 
-            SearchedString = query.ToLower();
+            SearchedString = query;
+        }
+
+        internal void GridTapped(Grid grid)
+        {
+            SelectedTodoGridHeight = grid.Height;
```

Comment:
to code behind!

Reply:
This function uses a service (_nativeViewService) to set the Grid in the right position (right above the original tapped grid).
I guess I used a bad name but I am not sure that I should move it into the code-behind. Or?

Reply:
Well, you *must* simply do this because you pass a `Xamarin.Forms.Grid` object as a parameter.

One of the main principles of MVVM is:
The ViewModel **must not ever** have a reference to the view.
This is not a "should not" - see ;-)

It is a great challenge to abstract as much view-related logic so that it can be tested (is not directly referencing any UI components), I know, but it's worthwile

Reply:
I moved everything to code-behind. :)
51f57ebc67228c6030cbdb72aad143483d3a7cbd